common/persistence/sql/sql_execution_store_test.go

```diff
@@ -2943,3 +2943,605 @@ func TestCreateFailoverMarkerTasks(t *testing.T) {
 		})
 	}
 }
+
+func TestGetWorkflowExecution(t *testing.T) {
+	testCases := []struct {
+		name      string
+		req       *persistence.InternalGetWorkflowExecutionRequest
+		mockSetup func(*sqlplugin.MockDB, *serialization.MockParser)
+		want      *persistence.InternalGetWorkflowExecutionResponse
+		wantErr   bool
+	}{
+		{
+			name: "Success case",
+			req: &persistence.InternalGetWorkflowExecutionRequest{
+				DomainID: "ff9c8a3f-0e4f-4d3e-a4d2-6f5f8f3f7d9d",
+				Execution: types.WorkflowExecution{
+					WorkflowID: "test-workflow-id",
+					RunID:      "ee8d7b6e-876c-4b1e-9b6e-5e3e3c6b6b3f",
+				},
+			},
+			mockSetup: func(db *sqlplugin.MockDB, parser *serialization.MockParser) {
+				db.EXPECT().SelectFromExecutions(gomock.Any(), gomock.Any()).Return([]sqlplugin.ExecutionsRow{
```

Comment:
It might be easier to use a mock sql like https://github.com/DATA-DOG/go-sqlmock rather than a gomocked entity.